main.nf

```diff
@@ -538,28 +542,30 @@ process SNPEFF {
 //----------------------------------------
 process SNIPPY {
     tag "$sample_id"
-    publishDir "${params.outdir}/variant/snippy",
+    publishDir "${params.outdir}/variant/snippy/$method",
                 pattern: "*snps.*", mode: 'copy'
-    publishDir "${params.outdir}/logs/${params.prediction}",
+    publishDir "${params.outdir}/logs/${params.prediction}/$method",
                 pattern: "${sample_id}.log", mode: "copy"
     
     input: 
-        tuple path(variant), path(ref), path(depths)
+        tuple val(method), path(variant), path(ref), path(depths)
         tuple val(sample_id), path(r1), path(r2)
 
     output:
         file "${sample_id}.log"
-        tuple val(sample_id), path(variant), path(depths),
+        tuple val(method), val(sample_id), path(variant), path(depths),
                 emit: 'annotation' 
     
     script:
-        outdir = "${params.outdir}/variant/snippy"
+        outdir = "${params.outdir}/variant/snippy/$method"
```

Comment:
Do not output directly to the output directory. Output to a relative path in the work directory created for the process with a name that uniquely identifies the output dir relative to other sample/reference genome combos.

```nextflow
publishDir "${params.outdir}/snippy", pattern: "snippy-*/"
...
output:
...
tuple sample_id, path(ref_fasta), path(snippy_outdir), emit: snippy

script:
ref_name = file(ref_fasta).getBaseName()
prefix = "${sample_id}-TO-${ref_name}"
snippy_outdir = "snippy-$prefix"
"""
snippy ... --outdir $snippy_outdir --prefix $prefix
"""
```

Also the Snippy log file should be in the Snippy output dir.

Reply:
I have added the prefix argument to SNIPPY process.